test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1111,6 +1110,53 @@ public async Task NullableScalar()
             }
         }
 
+        [Test, Description("Times out an async operation, testing that cancellation occurs successfully")]
+        public async Task ReaderTimeoutAsyncSoft()
```

Comment:
I think this test would be simpler (and possibly a bit more reliable) with the new PG mock, what do you think?

Reply:
I see your point - but PG mock isn't here yet 😢

Reply:
It is now :smile:

Reply:
Let me know if you have any issues implementing with the mock, the infrastructure is very new and may need to change a bit to accommodate new stuff.